src/NServiceBus.Core/Serialization/SerializationFeature.cs

```diff
@@ -18,6 +18,15 @@ public SerializationFeature()
             Defaults(s =>
             {
                 s.SetDefault("AdditionalDeserializers", new List<SerializationDefinition>());
+                var conventions = s.Get<Conventions>();
+                var knownMessages = s.GetAvailableTypes().Where(conventions.IsMessageType);
+
+                var messageMetadataRegistry = new MessageMetadataRegistry(conventions);
+                foreach (var msg in knownMessages)
+                {
+                    messageMetadataRegistry.RegisterMessageType(msg);
+                }
+                s.Set<MessageMetadataRegistry>(messageMetadataRegistry);
```

Comment:
It is not that easy. `Setup` is not allowed to change the settings. If we revert this back and use the container than at least some of the DI reductions in the `Routing` feature need to be reverted too. Not saying it is bad though. I agree this does not look like a typical default settings.

Maybe a compromise would be to set the `MessageMetadataRegistry` value here but do the type registration in `Setup`? That would work.